File: javalgo/exercise/Solution45.java
class Solution45 {
  public static void main(String[] args) {
    System.out.println(new Solution45().solution(13));
  }

  public String solution(int n) {
    StringBuffer answer = new StringBuffer();
    String[] num = { "1", "2", "4" };

    while (n >= 1) {
      answer.append(num[(n - 1) % 3]);
      n = (n - 1) / 3;
    }
    return answer.reverse().toString();
  }
}

File: python/kakao/Solution45.py
def solution(people, limit):
	answer = 0
	people = sorted(people, reverse=True)
	left = 0
	right = len(people) - 1
	while left <= right:
		if people[left] + people[right] <= limit:
			right -= 1
		left += 1
		answer += 1
	return answer

people = [70, 80, 50]
limit = 100
print(solution(people, limit))
